src/controller.cpp

```diff
@@ -185,38 +185,36 @@ void Controller::setPlan(geometry_msgs::Transform current_tf, geometry_msgs::Twi
   tf2::Transform deltaPlan;
   // We define segment0 to be the segment connecting pose0 and pose1.
   // Hence, when picking the starting path's pose, we mean to start at the segment connecting that and the next pose.
-  for (int idx_path = global_plan_tf_.size() - 2; idx_path >= 0; --idx_path)
+  for (auto idx_path = global_plan_tf_.size() - 1; idx_path > 0; --idx_path)
```

Comment:
Did you see https://github.com/nobleo/path_tracking_pid/issues/48? I'm worried about unsigned underflow. Especially with loops going backward

Reply:
Yes, you're right.

I accidentally changed the behavior of the code:
- it was incorrect if plan size was very large (>= INT_MAX + 3)
- it now is incorrect if the plan size is 0.
- 
The latter is more likely, so I'll change this back to an int and add a static cast to silence the warning.

Reply:
Fixed.